### PR TITLE
Add error handling to all boost filesystem functions

### PR DESCRIPTION
**Pull request by uhliksk**
All boost filesystem functions should be handled using error code to prevent random crashes caused by inaccesible files or directories. Increment in iterator should use separate error code variable because it have to be handled specifically to prevent infinite loop in diving to inaccessible directories.

To prevent coding errors in future the lint test is added to check if error code parameters are present.

Previous conversation about issue in #18095

### CODE REVIEW

File: src/dbwrapper.cpp [hunk 1]
```diff
@@ -133,7 +133,7 @@ CDBWrapper::CDBWrapper(const fs::path& path, size_t nCacheSize, bool fMemory, bo
             leveldb::Status result = leveldb::DestroyDB(path.string(), options);
             dbwrapper_private::HandleError(result);
         }
-        TryCreateDirectories(path);
+        assert(TryCreateDirectories(path));
```
Comment (practicalswift) [inline src/dbwrapper.cpp]: `assert(…)`:s should not have side-effects :)

**Review by practicalswift — COMMENTED** [review]
Concept ACK

What a great first-time contribution! Hope to see more contributions from you :)

File: src/dbwrapper.cpp (hunk 1 above)
Comment (uhliksk) [reply src/dbwrapper.cpp]: Yes, sorry, I was debugging and I forgot to handle this correctly. I'll add proper error handling here in additional commit.

**Review by promag — COMMENTED** [review]
Concept ACK.

File: src/init.cpp [hunk 2]
```diff
@@ -281,12 +281,13 @@ void Shutdown(NodeContext& node)
     }
 #endif
 
-    try {
-        if (!fs::remove(GetPidFile())) {
-            LogPrintf("%s: Unable to remove PID file: File does not exist\n", __func__);
-        }
-    } catch (const fs::filesystem_error& e) {
-        LogPrintf("%s: Unable to remove PID file: %s\n", __func__, fsbridge::get_filesystem_error_message(e));
```
Comment (MarcoFalke) [inline src/init.cpp:289]: this seems fine as is (catching the error), no?

**Comment by adamjonas** [comment]
Hi @uhliksk - pinging for rebase and response to [question above](https://github.com/bitcoin/bitcoin/pull/18189#discussion_r406800675).

**Comment by uhliksk** [comment]
Hi @adamjonas , sorry for delay, I'll be back on June 1st.

**Comment by adamjonas** [comment]
Hi @uhliksk, friendly monthly ping.

**Comment by uhliksk** [comment]
Hi @adamjonas, I'm alive but I had to postpone my work again. I'll be back soon.

**Comment by fanquake** [comment]
Going to close this for now. Let us know if/when you have time to work on it again and it can be re-opened.

**Pull request closed without merging**